### PR TITLE
Deletion job resources even in case configmap not found

### PR DESCRIPTION
**Pull request by omer-dayan**

### CODE REVIEW

File: pkg/workflow/workflow.go [hunk 1]
```diff
@@ -64,16 +67,81 @@ func DeleteJob(jobName string, namespaceInfo types.NamespaceInfo, clientset kube
...
+		log.Warningf("Delete configmap %s failed due to %v. Please clean it manually", configMapName, err)
 		log.Warningf("Please run `kubectl delete -n %s cm %s`", namespace, configMapName)
 		return err
 	}
 
 	return nil
 }
 
+func deleteJobResourcesWithoutConfigMap(jobName string, namespaceInfo types.NamespaceInfo, clientset kubernetes.Interface) error {
+	client, err := client.GetClient()
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:81]: can't we use clientset?

File: pkg/workflow/workflow.go [hunk 2]
```diff
@@ -64,16 +67,81 @@ func DeleteJob(jobName string, namespaceInfo types.NamespaceInfo, clientset kube
...
+	if err != nil {
+		return err
+	}
+
+	matchedJobs, err := trainer.GetTrainingJobsByTypeMap(jobName, namespaceInfo.Namespace, client)
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:86]: cant you use `SearchTrainingJob`

File: pkg/workflow/workflow.go [hunk 3]
```diff
@@ -64,16 +67,81 @@ func DeleteJob(jobName string, namespaceInfo types.NamespaceInfo, clientset kube
...
+	if err != nil {
+		return err
+	} else if len(matchedJobs) == 0 {
+		return cmdUtil.GetJobDoesNotExistsInNamespaceError(jobName, namespaceInfo)
+	} else if len(matchedJobs) > 1 {
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:91]: nice

File: pkg/workflow/workflow.go [hunk 4]
```diff
@@ -64,16 +67,81 @@ func DeleteJob(jobName string, namespaceInfo types.NamespaceInfo, clientset kube
...
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func deleteInteractiveJobResources(jobName, namespace string, clientset kubernetes.Interface) {
+	if _, err := clientset.AppsV1().StatefulSets(namespace).Get(jobName, metav1.GetOptions{}); err == nil {
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:125]: why use get first? why not just delete?

File: pkg/workflow/workflow.go [hunk 5]
```diff
@@ -64,16 +67,81 @@ func DeleteJob(jobName string, namespaceInfo types.NamespaceInfo, clientset kube
...
+		err := clientset.AppsV1().StatefulSets(namespace).Delete(jobName, &metav1.DeleteOptions{})
+		if err != nil {
+			log.Warnf(fmt.Sprintf("Failed to remove statefulSet %v, it may be removed manually and not by using Run:AI CLI.", jobName))
+		}
+	}
+	if _, err := clientset.CoreV1().Services(namespace).Get(jobName, metav1.GetOptions{}); err == nil {
```
Comment (yodarshafrir1) [inline pkg/workflow/workflow.go:131]: do we really need to delete the services and the ingress?

**Review by yodarshafrir1 — CHANGES_REQUESTED** [review]
please change this to debug as well:
```
if err != nil {
			log.Warnf("Failed to remove some of the job's resources, they might have been removed manually and not by using Run:AI CLI.")
		}

```

File: pkg/workflow/workflow.go (hunk 5 above)
Comment (omer-dayan) [reply pkg/workflow/workflow.go:131]: In case the job was interactive we need to delete it, why not?

File: pkg/workflow/workflow.go (hunk 2 above)
Comment (omer-dayan) [reply pkg/workflow/workflow.go:86]: `SearchTrainingJob` require to know the map firstly, to say "mpijob" or "runaijob"

File: pkg/workflow/workflow.go (hunk 1 above)
Comment (omer-dayan) [reply pkg/workflow/workflow.go:81]: clientset have an access to the basic k8s object like Statefulset/ ingress/ pod/ etc.
For being able to access to runaijob/ mpijob we need the extension to clientset, which build with client.

But for interactiveJobs we use the clientset

Comment (yodarshafrir1) [reply pkg/workflow/workflow.go:81]: please move this to mpi and runai then

File: pkg/workflow/workflow.go (hunk 2 above)
Comment (yodarshafrir1) [reply pkg/workflow/workflow.go:86]: please check

**Pull request closed without merging**